harvester/actor.py

```diff
@@ -80,14 +130,140 @@ def has_person_data(self):
         """
         return self._person_dict is not None
 
+    @property
+    def _organization_uri(self):
+        """
+        Return URI for the affiliation of this actor.
+
+        The URIs are from http://uri.suomi.fi/codelist/fairdata/organization/code. For
+        FIN-CLARIAH affiliations, the home organization from the `departmentName` field
+        is used when determining the URI.
+
+        Organization names seem to always available in English, so English names are
+        used for matching.
+
+        Raises UnknownOrganizationException if URI match is not found.
+        """
+
+        organization_name = self.data["affiliation"]["organizationName_en"]
+
+        if organization_name == "FIN-CLARIN":
+            organization_name = self.data["affiliation"]["departmentName_en"]
+
+        url_base = "http://uri.suomi.fi/codelist/fairdata/organization/code"
+        organization_codes = {
+            "Aalto University": "10076",
+            "CSC — IT Center for Science Ltd": "09206320",
+            "Centre for Applied Language Studies": "01906-213060",
+            "National Library of Finland": "01901-H981",
+            "South Eastern Finland University of Applied Sciences": "10118",
+            "University of Eastern Finland": "10088",
+            "University of Helsinki": "01901",
+            "University of Jyväskylä": "01906",
+            "University of Oulu": "01904",
+            "University of Tampere": "10122",
+            "University of Turku": "10089",
+        }
+        if organization_name in organization_codes:
+            return f"{url_base}/{organization_codes[organization_name]}"
+        raise UnknownOrganizationException(
+            f"Could not determine URI for {organization_name}"
+        )
+
+    def _none_if_no_affiliation(func):
+        """
+        Wrapper for properties that are None if the actor does not have affiliation data
+        """
+        # Wrapper shouldn't have self despite being defined in class
```

Comment:
Could this be replaced with `@staticmethod`?

Reply:
That does not play nicely with decoratorness:
```
(.venv) vagrant@kpdev:~/scratch/kielipankki-metax-bridge$ git diff harvester/actor.py
diff --git a/harvester/actor.py b/harvester/actor.py
index 90255bc..af7cf61 100644
--- a/harvester/actor.py
+++ b/harvester/actor.py
@@ -170,6 +170,7 @@ class Actor:
             f"Could not determine URI for {organization_name}"
         )
 
+    @staticmethod
     def _none_if_no_affiliation(func):
         """
         Wrapper for properties that are None if the actor does not have affiliation data
(.venv) vagrant@kpdev:~/scratch/kielipankki-metax-bridge$ python -m pytest
ImportError while loading conftest '/home/vagrant/scratch/kielipankki-metax-bridge/tests/conftest.py'.
tests/conftest.py:9: in <module>
    from harvester.metadata_parser import MSRecordParser
harvester/metadata_parser.py:7: in <module>
    from harvester.actor import Actor
harvester/actor.py:4: in <module>
    class Actor:
harvester/actor.py:191: in Actor
    def organization_name(self):
E   TypeError: 'staticmethod' object is not callable
```
Might be workaroundable (maybe even with something really simple, like `Actor._none_if_no_affiliation` but it feels silly to have a "private" static method anyway. The second option I considered was putting the wrapper outside the class altogether, but that also didn't feel good because the wrapper only works for this specific class with it's `self.data` dict.